src/main.rs

```diff
@@ -1,18 +1,161 @@
 use macroquad::prelude::*;
 
+const UPDATE_INTERVAL: f64 = 0.3;
+
+const SPACE_FOR_TEXT: i32 = 200;
+
 mod conways;
 
-#[macroquad::main("BasicShapes")]
+use conways::*;
+
+fn conf() -> Conf {
+    Conf {
+        window_title: "Conway's Game of Life".to_string(),
+        window_height: 600 + SPACE_FOR_TEXT,
+        window_width: 600,
+        fullscreen: false,
+        window_resizable: false,
+        ..Default::default()
+    }
+}
+
+#[macroquad::main(conf)]
 async fn main() {
+    let mut keys: [KeyCode; 10] = [KeyCode::Space; 10];
+    let mut offset = 0;
+
+    let mut grid = Grid::new(22);
+
+    // Glider 
+    grid.change_state((2, 1), State::Alive);
+    grid.change_state((2, 2), State::Alive);
+    grid.change_state((2, 3), State::Alive);
+    grid.change_state((1, 3), State::Alive);
+    grid.change_state((0, 2), State::Alive);
+
+    let mut last_updated = get_time();
+
+    let cube_width = 30.0;
+    let cube_height = 30.0;
+
+    let mut pause = true;
+
     loop {
-        clear_background(RED);
+        clear_background(GRAY);
+        if let Some(key) = get_last_key_pressed() {
+            keys[offset] = key;
+            offset += 1;
+            if offset == keys.len() {
+                offset = 0;
+            }
+            if key == KeyCode::P {
+                pause = !pause;
+            }
+        }
+
+        for x in 0..grid.get_height() {
+            for y in 0..grid.get_width() {
+                let color = match grid.get_state((x, y)) {
+                    State::Alive => BLACK,
+                    State::Dead => WHITE,
+                };
+
+                let offset_x = 0.0 + (cube_width + 0.0) * x as f32;
+                let offset_y = 0.0 + (cube_height + 0.0) * y as f32;
 
-        draw_line(40.0, 40.0, 100.0, 200.0, 15.0, BLUE);
-        draw_rectangle(screen_width() / 2.0 - 60.0, 100.0, 120.0, 60.0, GREEN);
-        draw_circle(screen_width() - 30.0, screen_height() - 30.0, 15.0, YELLOW);
+                draw_rectangle(offset_x, offset_y, cube_width, cube_height, color);
+            }
+        }
 
-        draw_text("IT WORKS!", 20.0, 20.0, 30.0, DARKGRAY);
+        if is_mouse_button_pressed(MouseButton::Left) {
+            let (mouse_x, mouse_y) = mouse_position();
+            let mouse_x = (mouse_x / cube_width).ceil() - 1.0;
+            let mouse_y = (mouse_y / cube_height).ceil() - 1.0;
```

Comment:
Maybe change this to `floor` instead of ceil so you don't have to subtract 1

Reply:
Changed here: https://github.com/lima-limon-inc/Conways-game-of-life/pull/3/commits/01e1d3e7b11d326fd7ca7a9886435c2282ae9a57

Thank you Lean ^_^